lib/puppet/util.rb

```diff
@@ -296,6 +296,7 @@ def which(bin)
   AbsolutePathWindows = %r!^(?:(?:[A-Z]:#{slash})|(?:#{slash}#{slash}#{label}#{slash}#{label})|(?:#{slash}#{slash}\?#{slash}#{label}))!io
   AbsolutePathPosix   = %r!^/!
   def absolute_path?(path, platform=nil)
+    return false unless path.is_a?(String)
```

Comment:
I don't think this is the best way to guard against the Ruby 2.7 warning about the use of `=~` on line 317. But, like I mentioned in the reply to your comment, there are multiple places that invoke `Pupet::Util.absolute_path?`. We'll hit the deprecation warning unless we clean up every place this is invoked. My gut says it's cleanest to put the guard here, but I also have hesitations.